nd4j-backends/nd4j-backend-impls/nd4j-native/src/main/java/org/nd4j/linalg/cpu/nativecpu/CpuSparseNDArrayFactory.java

```
@@ -25,6 +25,63 @@
 
     public CpuSparseNDArrayFactory(){}
 
+    @Override
+    public INDArray createSparse(double[] data, int[] columns, int[] pointerB, int[] pointerE, int[] shape){
+        return new SparseNDArrayCSR(data, columns, pointerB, pointerE, shape);
+    }
+
+    //  TODO ->
+
+    @Override
+    public IComplexFloat createFloat(float real, float imag) {
+        return null;
+    }
+
+    @Override
+    public IComplexDouble createDouble(double real, double imag) {
+        return null;
+    }
+
+    @Override
+    public IComplexNDArray createComplex(INDArray arr) {
+        return null;
+    }
+
+    @Override
+    public IComplexNDArray createComplex(IComplexNumber[] data, int[] shape) {
+        return null;
+    }
+
+    @Override
+    public IComplexNDArray createComplex(List<IComplexNDArray> arrs, int[] shape) {
+        return null;
+    }
+
+    @Override
+    public INDArray create(float[] data, int[] shape, int[] stride, int offset) {
+        return null;
+    }
+
+    @Override
+    public IComplexNDArray createComplex(double[] data, int[] shape, int[] stride, int offset) {
+        return null;
```

Same thing you can throw new UnsupportedoperationExample on the sparse stuff for now.

The create operations for things like the float arrays I'd imagine you could do a toSparse() maybe?